File: src/dotnet/PloyBus.Tests/Transport/Transactions/Messages/Handlers/Serializers/JsonHandlersTests.cs
using NUnit.Framework;

namespace PolyBus.Transport.Transactions.Messages.Handlers.Serializers;

[TestFixture]
public class JsonHandlersTests
{
    private JsonHandlers _jsonHandlers = null!;
    private Messages _messages = null!;

    [SetUp]
    public void SetUp()
    {
        _jsonHandlers = new JsonHandlers();
        _messages = new();
        _messages.Add(typeof(JsonHandlerTestMessage));
    }

    public async Task Serializer_SetsBodyAndContentType()
    {
        // Arrange
        var message = new JsonHandlerTestMessage { Text = "Hello, World!" };
        var outgoingMessage = new OutgoingMessage(null!, message)
        {
            Message = message,
            MessageType = typeof(JsonHandlerTestMessage)
        };
        var transaction = new OutgoingTransaction(null!);
        transaction.Add(message);

        // Act
        await _jsonHandlers.Serializer(transaction, () => Task.CompletedTask);

        // Assert
        Assert.That(outgoingMessage.Body, Is.Not.Null);
        Assert.That(outgoingMessage.Headers[Headers.ContentType], Is.EqualTo("application/json"));
    }
}

File: src/kotlin/src/test/kotlin/polybus/transport/transactions/messages/handlers/serializers/JsonHandlersTests.kt
package polybus.transport.transactions.messages.handlers.serializers

import kotlinx.coroutines.runBlocking
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test
import polybus.Headers
import polybus.PolyBusBuilder

class JsonHandlersTests {
    @Test
    fun serializerSetsBodyAndContentType() = runBlocking {
        // Arrange
        val jsonHandlers = JsonHandlers()
        val builder = PolyBusBuilder()
        builder.messages.add(JsonHandlerTestMessage::class.java)
        val bus = builder.build()
        val transaction = bus.createOutgoingTransaction()
        val message = JsonHandlerTestMessage().apply { text = "Hello, World!" }
        val outgoing = transaction.add(message)

        // Act
        jsonHandlers.serializer(transaction) {}

        // Assert
        assertTrue(outgoing.body.isNotBlank())
        assertTrue(outgoing.headers[Headers.CONTENT_TYPE] == "application/json")
    }
}
